### DISCUSSION TITLE
Is it possible to use serde_json with slint structs?

### DISCUSSION THREAD

**Discussion by j-piccolo** [slint-ui/slint#1563] [Q&A]
I assume probably not, if so, would you create another struct in rust and use that for serde_json and then copy that data to the slint struct or just map the values yourself to the slint struct?

**Comment by tronical** [comment]
Oh, adding a serde annotation is a good idea. I wonder if we could just `cfg_attr` it with `feature = "serde"` and once `serde` is an app dependency it'll become activated?

Meanwhile yes, you'll have to copy this by hand :(

@ogoffart What do you think? :)

**Reply by j-piccolo** [reply]
Would be nice to see, thanks for the response.

**Reply by ogoffart** [reply]
Yes that's a good idea.
We would need to 
 - generate a `#[cfg_attr(feature="serde", derive(...))]` for generated struct (in `generate_struct` in rust.rs)
 - Also derive these traits for our own types like `SharedString`, and co. (behind a feature flag)

**Reply by isaac-eiter** [reply]
Is it planned to add this feature? I've got a pretty complex/large struct that I'm currently importing manually via csv, and using json instead would be helpful when modifying that struct

**Reply by ogoffart** [reply]
We haven't planed to do it. But I created an issue to track it: https://github.com/slint-ui/slint/issues/2660
